specs/signing-and-verification-workflow.md

```diff
@@ -64,9 +64,53 @@ The user wants to pull an OCI artifact only if they are signed by a trusted publ
 1. **Get OCI artifact:** Using the verified digest, download the OCI artifact.
    This step is not in the purview of Notary Project.
+
+## Blob signing workflow
+
+The user wants to sign an arbitrary blob with a detached signature.
+
+### Signing Prerequisites
+
+- User has access to the signing certificate and private key or a remote signing service through a notation plug-in.
+
+### Signing Steps
+
+1. **Generate signature:** Using notation CLI or any other compliant signing tool, sign the blob. The signing tool should follow the following guideline.
+    1. Construct the blob payload as defined in [`signature specification`](./signature-specification.md#payload)
+    1. Verify that the signing certificate is valid and satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
+    1. Verify that the signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
+    1. Generate signature.
+        1. Generate signature using a signature format specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes). Also, as part of this step, the user-defined/supplied custom attributes should be added to the annotations of the signature payload.
+        1. If the user wants to timestamp the signature, obtain an [RFC-3161](https://datatracker.ietf.org/doc/html/rfc3161.html) compliant timestamp for the signature generated in the previous step. Otherwise, continue to the next step.
+            1. Verify that the timestamp signing certificate satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
+            1. Verify that the timestamp signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
+        1. Embed timestamp to the signature envelope.
+1. **Save the signature envelope:** Save the signature envelope generated in the previous step to a file. File extension should be the original blob file name plus `.sig.jws` for JWS signatures and `.sig.cose` for COSE signatures.
+
+## Arbitrary blob verification workflow
+
+The user wants to consume an arbitrary blob only if it was signed by a trusted publisher and the signature associated with the blob is valid.
+
+### Verification Prerequisites
+
+- User has the blob that they want to consume, along with its detached signature.
+- User has configured [trust store and trust policy](./trust-store-trust-policy.md) required for signature verification.
+
+### Verification Steps
+
+1. **Should implementations of this specification verify the signature? :** Depending upon [trust-policy](./trust-store-trust-policy.md#blob-trust-policy) configuration, determine whether implementations of this specification need to verify the signature or not.
+   If signature verification should be skipped for the given blob, skip the below steps and directly jump to step 4.
```

Comment:
Good catch. If the verification is skipped, then notation has no additional steps. Since the user already has the blob and they can choose to consume it (or not). In the OCI verification workflow, the "step 4" is downloading the OCI image.

I have updated the language accordingly.